### ISSUE TITLE
Support (`ReadOnly`)`Span` for buffer related operations

### ISSUE THREAD

**Issue by manison** [Pkcs11Interop/Pkcs11Interop#192]
Today all buffer related operations in both the high level and the low level APIs use arrays. When one must deal with external buffer slices this becomes inefficient and error prone since temporary arrays need to be allocated and content must be copied back and forth.

For example consider implementing the `ICryptoTransform.TransformBlock` method:

```cs
public int TransformBlock (byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset);
```

If the `inputOffset` parameter is non-zero, temporary array must be allocated and `inputCount` bytes must be copied from it before it can be passed to `C_EncryptUpdate` method. If a new overload would be added to `Pkcs11Library` supporting the `Span`

```cs
public CKR C_EncryptUpdate(NativeULong session, ReadOnlySpan<byte> part, Span<byte> encryptedPart, ref NativeULong encryptedPartLen);
```

the unnecessary allocation could be eliminated and the call would be simplified to

```cs
public int TransformBlock (byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
{
  ...
  _lib.C_EncryptUpdate(_session,
    new ReadOnlySpan<byte>(inputBuffer, inputOffset, inputCount),
    new Span<byte>(outputBuffer, outputOffset, outputBuffer.Length - outputOffset),
    ref len);
  ...
}
```

The downside of this is the `Span` could be only supported for net45 and netstandard2.0 builds through the System.Memory package.

**Comment by jariq** [comment]
Yes, `Span` utilization would be a great addition but as you already mentioned it would limit supported platforms. Maybe it will land in Pkcs11Interop 6 along with #160 and #161. We'll see.
